Sources/App/routes.swift

```diff
@@ -12,19 +12,21 @@ func routes(_ app: Application) throws {
     }
 
     /// Returns single product with details
-    app.get("product", ":id", "details") { req -> Product in
-        guard let id = req.parameters.get("id", as: UUID.self
-        ) else {
+    app.get("products", ":id") { req -> EventLoopFuture<Product> in
+        guard let id = req.parameters.get("id", as: UUID.self) else {
             throw Abort(.badRequest)
         }
 
-        guard let product = Stub().products.first(where: { product in
-            product.id == id
-        }) else {
-            throw Abort(.badRequest)
-        }
+        return Product.find(id, on: req.db)
+            .unwrap(or: Abort(.notFound))
+    }
+
+    let product = app.grouped("product")
 
-        return product
+    // Create a single product to post to the DB
+    product.post("submit") { req -> EventLoopFuture<Product> in
```

Comment:
I am a little confused with this single product submit API. When do we use it?
From the app side, we submit the entire `ShoppingCart` object. After tap the submit button.
Even for the save feature, the app should submit the `ShoppingCart` object, instead of making multiple single product calls right? 

Please let me know what I missed here

Reply:
Sorry I should have wrapped this in a DEBUG flag. This is simply adding a single product via Postman. I was testing getting products added to the database and this was a way to test routes.

We can either remove this or wrap it in a debug flag. What do you think?

Reply:
Haha. That makes sense. Don't worry about it. We can remove it later.